biocrnpyler/components_combinatorial_conformation.py

```diff
@@ -0,0 +1,504 @@
+                active_state = False
+                if  (s in self.promoter_states and self.promoter_states_on) or (s not in self.promoter_states and not self.promoter_states_on):
+                    active_state = True
+
+                active_complex = False
+                if any([s.get_complex(c) is not None for c in self.activating_complexes]):
+                    active_complex = True
+
+                innactive_complex = False
+                if any([s.get_complex(c) is not None for c in self.inactivating_complexes]):
+                    innactive_complex = True
+                    if (active_state and len(self.promoter_states)>0) or active_complex:
+                        warnings.warn("innactive_complex conflicts with active_complex or active_state in CombinatorialConformationPromoter. Defaulting to innactive.")
+                        
+                if (active_state or active_complex) and not innactive_complex:
+                    self.dna_to_bind = s.polymers[0][self.promoter_location]
+                    self.name = self._get_part_id(s) #Reset name for unique addressable promoter states
+                    promoter_species += Promoter.update_species(self)
+        
+        return list(set(promoter_species+self.conformation_species))
+
+    def update_reactions(self):
+        if not hasattr(self, "conformation_species"):
+            self.update_species
+
+        conformation_reactions = CombinatorialConformation.update_reactions(self)
+        promoter_reactions = []
+        old_name = self.name
+        for s in self.conformation_species:
+            if isinstance(s, PolymerConformation):
+                active_state = False
+                if  (s in self.promoter_states and self.promoter_states_on) or (s not in self.promoter_states and not self.promoter_states_on):
+                    active_state = True
+
+                active_complex = False
+                if any([s.get_complex(c) is not None for c in self.activating_complexes]):
+                    active_complex = True
+
+                innactive_complex = False
+                if any([s.get_complex(c) is not None for c in self.inactivating_complexes]):
+                    innactive_complex = True
+                    if (active_state and len(self.promoter_states)>0) or active_complex:
+                        warnings.warn("innactive_complex conflicts with active_complex or active_state in CombinatorialConformationPromoter. Defaulting to innactive.")
+                        
+                if (active_state or active_complex) and not innactive_complex:
+                    self.dna_to_bind = s.polymers[0][self.promoter_location]
+                    self.name = self._get_part_id(s) #Reset name for unique addressable promoter states
+                    promoter_reactions += Promoter.update_reactions(self)
+        self.name = old_name
+        return promoter_reactions + conformation_reactions
```

Comment:
I don't think i've fully understood this but it seems useful

Reply:
The idea is that I reuse a Promoter by changing its internal "DNA_to_bind" (instead of enumerating many promoters like you do in DNA construct). This allows many species (conformations) to reuse the same promoter object. It is a little hacky, but because all the promoters make the same ultimate transcript I think it is okay.